src/OpenTelemetrySDK/OpenTelemetryController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+using LogLevel = Microsoft.Azure.WebJobs.Script.Grpc.Messages.RpcLog.Types.Level;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.OpenTelemetrySDK
+{
+    using PowerShell = System.Management.Automation.PowerShell;
+
+    internal class OpenTelemetryController
+    {
+        private readonly ILogger _logger;
+        private PowerShell _pwsh { get; set; }
```

Comment:
The direct dependency on the `PowerShell` class will complicate unit testing. Consider taking a dependency on `IPowerShellServices` instead. You may want to add either the `AddCommand` method to `IPowerShellServices`, or (probably better) more specific methods for each AddCommand/AddParameter/InvokeAndClearCommands batch.

Reply:
The structure of the Durable folders in the repo, and their associated tests, is quite a bit different (and more complex) than the current implementation of OpenTelemetry. In Durable, the DurableController is a high-level object that the other components in the worker interact with to call durable methods, but under the hood it uses it's own IPowerShellServices object (which is a durable-specific object defined in the DurableSDK folder, so I very much object to the naming there - it probably should be called IDurablePowerShellServices or maybe just IDurableServices, but what can we do) to actually do the work using the PowerShell and ILogger objects passed to it in its constructor.  

In OpenTelemetry, as it was implemented at the time of your review, the OpenTelemetryController acted as both the controller that the worker interacts with and the provider of the services underneath. 

I have refactored this - to more closely match the Controller/IServices/Services pattern, which better supports testing.

Reply:
Right, I haven't realized that IPowerShellServices is used exclusively for Durable. Creating a new interface is a good idea.